### ISSUE TITLE
First Changes: Images/Captions

### ISSUE THREAD

**Issue by vhellstein** [DenverArtMuseum/biophilia-pub#22]

**Comment by vhellstein** [comment]
- [ ] in the essays: captions include designer, work, date, and collection: @vhellstein can take care of this 


- [ ] in the lightbox: full image captions, but don’t overlap with image. See https://weisberg-main.netlify.app/essays/essay-smith/ 

- [ ] move full screen icon to upper right of image, with zoomable icons beneath that 

- [ ] make sure image carousel arrows are correct, e.g., no left arrow on first image, no right arrow on last image in carousel. Don’t wrap around: @1000camels, can we fix this? If it too involved, let's reassess

- [ ] Can we have individual carousels on one page? (For those objects that have multiple views): @1000camels, is this possible? If too hard, we can find a different solution. [Newfields](https://paintings.theclowescollection.org/building-the-clowes-collection/) doesn’t have a page carousel, but not sure this is the same as what we’re wanting. 

- [ ] If we can figure out the individual carousels, Darrin would like to have a few images break the column of text (like the quotes at the beginning of his introduction). These will be identified in next round. 

- [ ] no lines surrounding embedded images. See [Perfectly Imperfect](https://buncheong.denartmus.org/beyond-borders/) 

- [ ] images should come after figure call outs— @vhellstein will take care of this

**Comment by vhellstein** [comment]
@1000camels I know you've been working on the mobile version, but can we prioritize sorting some of these image things that are still outstanding? I was chatting with @mpopke about zoomable images in the Object Plates, and he said that would be easiest for you to do. @mpopke is also going to reach out to the Quire folks about overriding in line figure captions so I can work on that ASAP.

**Comment by 1000camels** [comment]
I added zoom: true to all figures that had images greater than 512px (required). It complains otherwise. So if we improve any of the images, we'll need to add that.

The images are zooming, but the zoom buttons were not enabled by default. I tried to enable them but their styling is bad and the code to zoom is not fully functioning yet. Let me know if you want me to try to get them to work.

**Comment by 1000camels** [comment]
Also, I do not know much about the IIIF images. I am happy to try to implement those, but it sounds like it requires a bunch of preprocessing of the images. @mpopke what do you think?

**Comment by 1000camels** [comment]
@vhellstein I do not think it is easy to hide the arrow at the start and end. But I will continue to look into it

**Comment by vhellstein** [comment]
Okay, we might make this a lower priority then.

**Comment by vhellstein** [comment]
> I added zoom: true to all figures that had images greater than 512px (required). It complains otherwise. So if we improve any of the images, we'll need to add that.
> 
> The images are zooming, but the zoom buttons were not enabled by default. I tried to enable them but their styling is bad and the code to zoom is not fully functioning yet. Let me know if you want me to try to get them to work.

I think it would be good to have something to indicate that you can zoom. @mpopke thoughts?

**Comment by mpopke** [comment]
I also just realized I put the icons in the wrong corner. I can fix that today.

**Comment by vhellstein** [comment]
Following up on a few things from above

- [ ] @1000camels can we make each image have it's own image carousel, so not all of the images on the page are in the carousel? Newfields does this. Ideally, clicking an image embedded in an essay will open it into full screen with its accompanying full caption, and zoomability. But I'm waiting to hear from Darrin about zoomability of images in the essays, since Quire renders that visible image on the page at a little less resolution than the actual image file.
- [ ] @vhellstein to add hero image captions in the md files
- [ ] @vhellstein will shorten captions on essay page
- [ ] @mpopke will enable captions under videos
- [ ] @1000camels When an image pops up in the lightbox, we'd like to make sure it doesn't overlap the image at all. Right now, if the caption goes to 2 lines, there's still some overlap. Can we force the caption space to always 2 lines? Or will this always be an issue with responsive design?
- [ ] @1000camels and @mpopke Lightbox buttons need work across the board, some of them seemed broken. Also need a prompt "Press Esc. to exit lightbox/full screen mode" (or something like that)
- [ ] @1000camels Now that the lines around the embedded images are gone, can we align the top of the image with the text?


![Image](https://github.com/DenverArtMuseum/biophilia-pub/assets/119438504/0bbf2e33-e2e6-466d-b52e-66de8bf725ae)

**Comment by vhellstein** [comment]
@mpopke, I added captions to the videos in figure.yaml, so I think that should take care of the above?

@mpopke and @1000camels: In the lightboxes for the zoomable images, is it possible to remove the caption in its resting state, but have the caption appear when you enter full screen mode? If not, Darrin would like to remove captions from the lightboxes.

**Comment by 1000camels** [comment]
Newfields uses Open Sea Dragon (https://openseadragon.github.io/) to do their zoom, instead of Lightbox

**Comment by 1000camels** [comment]
I am working on the spacing of the images in the entries. I have pushed it up

**Comment by vhellstein** [comment]
> I am working on the spacing of the images in the entries. I have pushed it up

Maybe this hadn't made it in there yet, but the images looked even lower than before in the essays when I looked this morning. Maybe I'm misremembering though.

**Comment by vhellstein** [comment]
> I am working on the spacing of the images in the entries. I have pushed it up

Matt worked on this, too, and it looks good now.

**Comment by vhellstein** [comment]
> Newfields uses Open Sea Dragon (https://openseadragon.github.io/) to do their zoom, instead of Lightbox

Okay. Matt is also going to look into this. I don't know much, but I'm not sure using a new lightbox at this point is feasible.

**Comment by 1000camels** [comment]
I am sure it would be possible to get this to work, but it might take a day or so

**Comment by mpopke** [comment]
@1000camels @vhellstein I'm making some progress on this. Quire's lightbox, modal, and canvas-panel take some figuring out, but I've got the bones of a multi-slideshow working on my system now. I think I'll also be able to create a shortcode that allows us to create the slideshows Darrin was requesting that show images that aren't included inline in the essay body. I won't have it done this afternoon, but I can wrap it up tomorrow from home. 

Just letting you know so Darcy and I aren't duplicating effort, and Valerie can tell Darrin it's being worked on.

**Comment by 1000camels** [comment]
That's great.

**Comment by mpopke** [comment]
Changed lightbox modals so that we can have more than one per page. 

- You can specify a modal_id property in figures.yaml entries. The modal_id must be a valid HTML id string. I've found that Quire doesn't like underscores in these ids, but that's not a huge surprise because it doesn't seem to like underscores in any id.
- Any figures that do not have a modal_id specified will end up in the default modal lightbox for the page (basically the same behavior as before).
- Figures with a specified modal_id will **not** be in the default modal lightbox but will end up in a different lightbox identified by the modal_id
- Any figures on the same page that share the same modal_id will appear in the same modal lightbox. This allows us to put figuregroups all in the same lightbox, which seems like something we should do.

I already went through all of our figures to give them modal_id when appropriate. I am grouping figuregroups into the same lightbox, and I sometimes group other images if they're very close to each other and represent the same object (or group of objects by the same designer). It's easy to break up these shared lightboxes if we need to, but I think this is a pretty good setup so far.

The modals do not currently allow us to include images that are not shown inline in the text of the page, but I could easily add that functionality if we want to. Gonna talk to Valerie about that today.
